### ISSUE TITLE
Refactor nextflow tower and rnafusion classes

### ISSUE THREAD

**Issue by fevac** [Clinical-Genomics/cg#1819]
## Description

Refactor NextflowAnalysisAPI, TowerAnalysisAPI and RnafusionAnalysisAPI to make them cleaner and more general, particularly thinking about new upcoming pipelines as taxprofiler. 

## Suggested solution

*Write here if there is a suggested way to solve the issue*

## This can be closed when

*Describe what needs to be done for this issue to be closed*

## Blocked by

*If there are any blocking issues/prs/things in this or other repos. Please link to them.*

**Comment by fevac** [comment]
Duplicated. Closing in favour of https://github.com/Clinical-Genomics/cg/issues/2044
